events_structs.go

```diff
@@ -1062,10 +1064,62 @@ func (ev *EventDeleteNotifierType) String() string {
 	return fmt.Sprintf("Notifier '%s' deleted on %s app", d.NotifierName, ev.AppName)
 }
 
+// Edit hds_contact
+type EventEditHDSContactTypeData struct {
+	Name           string `json:"name"`
+	Email          string `json:"email"`
+	PhoneNumber    string `json:"phone_number"`
+	Company        string `json:"company"`
+	AddressLine1   string `json:"address_line1"`
+	AddressLine2   string `json:"address_line2"`
+	AddressCity    string `json:"address_city"`
+	AddressZip     string `json:"address_zip"`
+	AddressCountry string `json:"address_country"`
+	Notes          string `json:"notes"`
+}
+
+type EventEditHDSContactType struct {
+	Event
+	TypeData EventEditHDSContactTypeData `json:"type_data"`
+}
+
+func (ev *EventEditHDSContactType) String() string {
+	d := ev.TypeData
+	return fmt.Sprintf("Contact health Professional '%s' edited on %s app", d.Name, ev.AppName)
+}
+
+// Create data_access_consent
+type EventCreateDataAccessConsentTypeData struct {
+	EndAt      time.Time `json:"end_at"`
+	Databases  bool      `json:"databases"`
+	Containers bool      `json:"containers"`
+}
+
+type EventCreateDataAccessConsentType struct {
+	Event
+	TypeData EventCreateDataAccessConsentTypeData `json:"type_data"`
+}
+
+func (ev *EventCreateDataAccessConsentType) String() string {
+	d := ev.TypeData
+	res := "Additional access "
+	if d.Containers {
+		res += "to application runtime environment, "
+	} else if d.Databases {
```

Comment:
Isn't it possible that this event is about the consent of both the databases and containers? I.e. is it possible that both `Containers` and `Databases` are `true`?

Reply:
Oh yes you're right I have to create a third condition. Thanks.

Reply:
Or @EtienneM with just an other "if", I guess it would work.
```
res := "Additional access "
	if d.Containers {
		res += "to application runtime environment, "
	} 
	if d.Databases {
		res += "to databases metadata and monitoring data, "
	}
	res += fmt.Sprintf("created on %s app", ev.AppName)
	return res
```

Reply:
Well, it depends on what you want to display. If both are true, with your latter suggestion, it would display:

```
Additional access to application runtime environment, to databases metadata and monitoring data, created on my-app app
```

If that's what you want, then it's a good solution :)

Reply:
Yep it's what I wanted, guess I was tired when added the else if rather than if.